server/webapp/WEB-INF/rails.new/spec/views/admin/users/_user_results_table_html_spec.rb

```diff
@@ -28,4 +28,28 @@
 
     expect(response.body).to have_selector("input[type='radio'][name='selection'][id*='selection_button_user_username_dot_with_dot_dot']")
   end
+
+  it "should santize all fields when generating a DOM ID, to handle fields with special characters" do
+    allow(view).to receive(:random_dom_id) {|prefix| prefix + 'RAND'}
+
+    user_search_model = double('user_search_model')
+    user = User.new('!username-with_special_chars._@#$%^&*()-_ and spaces', '!display-name-with_special_chars_@#$%^&*()-_', "email!me#now*@dot.com")
+    user_search_model.should_receive(:getUser).at_least(:once).and_return(user)
+    user_search_model.should_receive(:getUserSourceType).and_return(com.thoughtworks.go.presentation.UserSourceType::LDAP)
+
+    render :partial => "admin/users/user_results_table.html", :locals => {:scope => { :users => [user_search_model]}}
```

Comment:
Line is too long. [117/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.